service/frontend/adminHandler.go

```diff
@@ -192,6 +204,52 @@ func (adh *AdminHandler) ListNamespaces(
 	}, err
 }
 
+// RegisterNamespace creates a new namespace which can be used as a container for all resources.  Namespace is a top level
+// entity within Temporal, used as a container for all resources like workflow executions, taskqueues, etc.  Namespace
+// acts as a sandbox and provides isolation for all resources within the namespace.  All resources belongs to exactly one
+// namespace.
+func (adh *AdminHandler) RegisterNamespace(ctx context.Context, request *adminservice.RegisterNamespaceRequest) (_ *adminservice.RegisterNamespaceResponse, retError error) {
+	defer log.CapturePanic(adh.GetLogger(), &retError)
+
+	scope, sw := adh.startRequestProfile(metrics.AdminRegisterNamespaceScope)
+	defer sw.Stop()
+
+	if request == nil {
+		return nil, adh.error(errRequestNotSet, scope)
+	}
+
+	if request.GetNamespace() == "" {
+		return nil, adh.error(errNamespaceNotSet, scope)
+	}
+
+	if len(request.GetNamespace()) > adh.config.MaxIDLengthLimit() {
+		return nil, adh.error(errNamespaceTooLong, scope)
+	}
+
+	req := &workflowservice.RegisterNamespaceRequest{
+		Namespace:                        request.GetNamespace(),
+		Description:                      request.GetDescription(),
+		OwnerEmail:                       request.GetOwnerEmail(),
+		WorkflowExecutionRetentionPeriod: request.GetWorkflowExecutionRetentionPeriod(),
+		Clusters:                         request.GetClusters(),
+		ActiveClusterName:                request.GetActiveClusterName(),
+		Data:                             request.GetData(),
+		SecurityToken:                    request.GetSecurityToken(),
+		IsGlobalNamespace:                request.GetIsGlobalNamespace(),
+		HistoryArchivalState:             request.GetHistoryArchivalState(),
+		HistoryArchivalUri:               request.GetHistoryArchivalUri(),
+		VisibilityArchivalState:          request.GetVisibilityArchivalState(),
+		VisibilityArchivalUri:            request.GetVisibilityArchivalUri(),
+	}
```

Comment:
you may want to simplify this

Reply:
discussed directly. There is some refactoring work on api - business-logic - persistence that we may want to do though separately from this PR